### PR TITLE
build fixes for freebsd

### PR DESCRIPTION
**Pull request by sambler**
I need to make these small changes to compile on FreeBSD.  The first two are #if based to only effect FreeBSD.

I would expect adding the SAMPLERATE_INCLUDE_DIRS to plugins/CMakeLists.txt would be needed for any platform.

In plugins/zynaddsubfx/CMakeLists.txt using CMAKE_DL_LIBS instead of -ldl is better for cross platform building. Though I'm not sure if this will stick with zynaddsubfx updates.

### CODE REVIEW

**Comment by tresf** [comment]
Thanks!!

@Lukas-W seems to have taken over as our CMake resident expert, so he should probably be the one to merge. :+1:

File: include/AudioPortAudio.h [hunk 1]
```diff
@@ -45,7 +45,11 @@ public slots:
 
 #ifdef LMMS_HAVE_PORTAUDIO
 
+#if defined(__FreeBSD__)
```
Comment (Lukas-W) [inline include/AudioPortAudio.h]: Can you try adding 
```cmake
INCLUDE_DIRECTORIES(${PORTAUDIO_INCLUDE_DIRS})`
```
to `CMakeLists.txt` instead?

File: include/lmms_math.h [hunk 2]
```diff
@@ -34,7 +34,7 @@
 #include <cmath>
 using namespace std;
 
-#if defined (LMMS_BUILD_WIN32) || defined (LMMS_BUILD_APPLE) 
+#if defined (LMMS_BUILD_WIN32) || defined (LMMS_BUILD_APPLE)  || defined (__FreeBSD__)
```
Comment (Lukas-W) [inline include/lmms_math.h]: Maybe this `#if` condition can be removed altogether?

**Comment by tresf** [comment]
@sambler please address Lukas' question and rebase this PR against upstream. :+1:

File: include/lmms_math.h (hunk 2 above)
Comment (Lukas-W) [inline include/lmms_math.h]: Please disregard that suggestion.

**Comment by tresf** [comment]
@sambler once you rebase this PR, we can merge.  Sorry for the inconvenience.

**Comment by tresf** [comment]
@sambler, again, if you have time to rebase this PR, we'll merge it in for you. :+1:

**Comment by tresf** [comment]
Actually... @curlymorphic if you have some time, these fixes should really go in and they're tremendously simple.  If you have time, could you patch them into a new PR so we can close this out?  We really need to get 1.2 ready. :+1:

**Comment by curlymorphic** [comment]
sure

**Pull request closed without merging**